### ISSUE TITLE
How can i do continued pre-training using this?

### ISSUE THREAD

**Issue by Aloukik21** [jiaweizzhao/GaLore#21]

**Comment by jiaweizzhao** [comment]
For now, you can specify your checkpoint path using `args.continue_from` in torchrun_main.py

**Comment by jjhoow** [comment]
Taking advantage of the question, I would like to know if I can use galore_adamw8bit_per_layer to train specific layers while freezing others? If so, could I use llama-pro (https://github.com/TencentARC/LLaMA-Pro) to increase the layers of a model like Mistral and then train only those layers using Galore?

**Comment by jiaweizzhao** [comment]
@jjhoow per-layer GaLore should achieve it out-of-box, as it will assign the optimizer only if param.requires_grad is True, see here: https://github.com/jiaweizzhao/GaLore?tab=readme-ov-file#save-weight-gradient-memory-using-per-layer-weight-updates

**Comment by jjhoow** [comment]
@jiaweizzhao I performed the test earlier with the code above and using galore_adamw8bit_per_layer and the code above, I need to get the parameters right because I noticed a roller coaster movement.
